pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -325,12 +327,18 @@ func (inClusterClientConfig) ClientConfig() (*client.Config, error) {
 }
 
 func (inClusterClientConfig) Namespace() (string, error) {
-	// TODO: generic way to figure out what namespace you are running in?
-	// This way assumes you've set the POD_NAMESPACE environment variable
-	// using the downward API.
+	// This way assumes you've set the POD_NAMESPACE environment variable using the downward API.
```

Comment:
This should have been prefixed with KUBERNETES_, KUBE_, or somesuch. :-(

Lots of examples use this, but I don't see it actually documented somewhere.